tests/test_testing.py

```diff
@@ -158,3 +158,70 @@ def test_pytest_frozen_time_tz_windows_pass(pytester, book_list_html_response) -
         2022, 3, 4, 20, 21, 22, tzinfo=dateutil.tz.tzlocal()
     )
     _assert_frozen_item(frozen_time, pytester, book_list_html_response)
+
+
+@attrs.define
+class ClientPage(WebPage):
+    client: HttpClient
+
+    async def to_item(self) -> dict:  # noqa: D102
+        resp1 = await self.client.get("http://books.toscrape.com/1.html")
+        resp2 = await self.client.get("http://books.toscrape.com/2.html")
+        return {"foo": "bar", "additional": [resp1.body.decode(), resp2.body.decode()]}
+
+
+def test_httpclient(pytester, book_list_html_response) -> None:
+    body1 = HttpResponseBody(b"body1")
+    url1 = "http://books.toscrape.com/1.html"
+    request1 = HttpRequest(url1)
+    response1 = HttpResponse(url=url1, body=body1, encoding="utf-8")
+    body2 = HttpResponseBody(b"body2")
+    url2 = "http://books.toscrape.com/2.html"
+    request2 = HttpRequest(url2)
+    response2 = HttpResponse(url=url2, body=body2, encoding="utf-8")
+    responses = [
+        (request1, response1),
+        (request2, response2),
+    ]
+    client = HttpClient(responses=responses)
+
+    base_dir = pytester.path / "fixtures" / get_fq_class_name(ClientPage)
+    item = {
+        "foo": "bar",
+        "additional": ["body1", "body2"],
+    }
+    Fixture.save(base_dir, inputs=[book_list_html_response, client], item=item)
```

Comment:
I think you can simplify it a bit using _save_fixture utility function added in #129.

Reply:
Why is using _save_fixture not possible here?

Reply:
ah, I see :)